src/devices/video/hd44780.cpp

```diff
@@ -254,12 +360,10 @@ TIMER_CALLBACK_MEMBER(hd44780_device::blink_tick)
 //  HELPERS
 //**************************************************************************
 
-void hd44780_device::set_busy_flag(uint16_t usec)
+void hd44780_device::set_busy_flag(uint16_t cycles)
 {
 	m_busy_flag = true;
-
-	usec = float(usec) * m_busy_factor + 0.5;
-	m_busy_timer->adjust(attotime::from_usec(usec));
+	m_busy_timer->adjust(attotime::from_hz(double(clock() / cycles) * m_busy_factor));
```

Comment:
This will invert the result with busy_factor?
eg. busy factor 0.5, will make the busy flag duration twice longer, or 0.0 would mean 0Hz which is attotime::never.

Reply:
This should be fixed now, although I'm going to see if I can completely remove the whole busy factor system in favor of just adjusting the clock for the few drivers that use it, in the future

Reply:
Nothing does, no.
And if I can make busy_factor go away completely, this will be a non-issue.